prymer/primer3/primer3.py

```diff
@@ -760,3 +793,30 @@ def _create_design_region(
         )
 
         return design_region
+
+
+def _check_design_results(design_input: Primer3Input, design_results: dict[str, str]) -> int:
+    """Checks for any additional Primer3 errors and reports out the count of emitted designs."""
+    count_tag = design_input.task.count_tag
+    maybe_count: Optional[str] = design_results.get(count_tag)
+    if maybe_count is None:  # no count tag was found
+        if "PRIMER_ERROR" in design_results:
+            primer_error = design_results["PRIMER_ERROR"]
+            raise ValueError(f"Primer3 returned an error: {primer_error}")
+        else:
+            raise ValueError(f"Primer3 did not return the count tag: {count_tag}")
+    count: int = int(maybe_count)
+
+    return count
+
+
+def _has_acceptable_dinuc_run(
+    design_input: Primer3Input, oligo_design: Union[Primer, Probe]
+) -> bool:
```

Comment:
```suggestion
) -> bool:
    """
    True if the design's longest dinucleotide run is no more than the stipulated maximum.
    
    For primer designs, the maximum is recorded in the input's `PrimerAndAmpliconParameters.primer_max_dinuc_bases`.
    
    For probe designs, the maximum is recorded in the input's `ProbeParameters.probe_max_dinuc_bases`.
```

**suggestion** Document what makes a dinucleotide run acceptable

(args/return blocks aren't necessary in private functions, but it's helpful to at least have a summary docstring)